app/ts/components/TokenManager/EtherBalance.tsx

```diff
@@ -0,0 +1,50 @@
+
+import { useSignalEffect } from '@preact/signals'
+import { BigNumber, ethers } from 'ethers'
+import { useAsyncState } from '../../library/preact-utilities.js'
+import { useAccount } from '../../store/account.js'
+import { useNetwork } from '../../store/network.js'
+import { useProviders } from '../../store/provider.js'
+import { AsyncText } from '../AsyncText.js'
+
+export const EtherBalance = () => {
+	const { network } = useNetwork()
+	const { address } = useAccount()
+	const providers = useProviders()
+	const { value: query, waitFor } = useAsyncState<BigNumber>()
+
+	const getBalance = () => {
+		if (address.value.state !== 'resolved') return
+		const accountAddress = address.value.value
+		waitFor(async () => {
+			const provider = providers.getbrowserProvider()
+			return await provider.getBalance(accountAddress)
+		})
+	}
+
+	useSignalEffect(() => {
+		if (network.value.state !== 'resolved' || address.value.state !== 'resolved') return
+		getBalance()
+	})
```

Comment:
Consider inlining `getBalance` into the `useSignalEffect`.  This would save you from having to check `address.value.state` twice.  Alternatively, if you like having a named function consider moving 100% of the `useSignalEffect` logic into `getBalance`.

Reply:
Same comment for token balance component below.